src/legend/interactiveLegend.ts

```diff
@@ -210,58 +216,60 @@ export class InteractiveLegend implements ILegend {
      */
     private ensureLegendTableCreated(): void {
         if (this.legendContainerDiv.select("div table").empty()) {
-            let legendTable: Selection<any> = this.legendContainerDiv
+            let legendTable: d3.Selection<any, any, any, any> = this.legendContainerDiv
                 .append("div")
                 .append("table");
 
             legendTable.style("table-layout", "fixed").append("tbody");
             // Setup Pan Gestures of the legend
-            this.setPanGestureOnLegend(legendTable);
+
+            // this.setPanGestureOnLegend(legendTable);
         }
     }
 
     /**
      * Set Horizontal Pan gesture for the legend
      */
-    private setPanGestureOnLegend(legendTable: Selection<any>): void {
-        let parentNode = <HTMLElement>this.legendContainerParent.node();
-        let viewportWidth: number = parentNode.getBoundingClientRect().width;
-        let xscale: Linear<number, number> = linear()
-            .domain([0, viewportWidth])
-            .range([0, viewportWidth]);
-
-        let legendZoom: Zoom<any> = zoom()
-            .scaleExtent([1, 1]) // disable scaling
-            .x(xscale)
-            .on("zoom", () => {
-                // horizontal pan is valid only in case the legend items width are bigger than the viewport width
-                if ($(legendTable[0]).width() > viewportWidth) {
-                    let t: number[] = legendZoom.translate();
-                    let tx: number = t[0];
-                    let ty: number = t[1];
-
-                    tx = Math.min(tx, 0);
-                    tx = Math.max(tx, viewportWidth - $(legendTable[0]).width());
-                    legendZoom.translate([tx, ty]);
-
-                    legendTable.style("-ms-transform", () => { /* IE 9 */
-                        return translateXWithPixels(tx);
-                    });
-
-                    legendTable.style("-webkit-transform", () => { /* Safari */
-                        return translateXWithPixels(tx);
-                    });
-
-                    legendTable.style("transform", () => {
-                        return translateXWithPixels(tx);
-                    });
-                }
-            });
-
-        if (this.legendContainerDiv) {
-            this.legendContainerDiv.call(legendZoom);
-        } else {
-            legendTable.call(legendZoom);
-        }
+    private setPanGestureOnLegend(legendTable: d3.Selection<any, any, any, any>): void {
+        throw "Not implmented";
```

Comment:
`Not implemented`.

By the way, what code calls it?

Reply:
It calls in ensureLegendTableCreated method. But it requires converting code related to using d3.ZoomBehavior. I don't know how to properly rewrite setPanGestureOnLegend with d3 v5.